### PR TITLE
Cucumber-data-table-documentation

### PR DESCRIPTION
**Pull request by ron-brosh**
This PR adds documentation to the ReadMe file about the DataTable usage followed by examples.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,6 +125,57 @@ Feature: Sample
...
+While the `DataTable`-like syntax is a good practice for scenarios that require repeated steps, for example, entering text in different fields, sometimes we want to prepare test data in a readable way and mock our scneario's prerequisites and assert the expected result in an explicit domain driven way.
+To handle this, we create a data table:
+```ruby
+Feature: Search songs
+
+  Scenario: Searched text matches a song's details
+    Given available songs
+    | 'artist'      | 'name'                      |
+    | 'The doors'   | 'Riders on the storm'       |
+    | 'Bob dylan'   | "Knockin' On Heaven's Door" |
```
Comment (olexale) [inline README.md:137]:
```suggestion
    | 'The Doors'   | 'Riders on the storm'       |
    | 'Bob Dylan'   | "Knockin' On Heaven's Door" |
```

File: README.md [hunk 2]
```diff
@@ -125,6 +125,57 @@ Feature: Sample
...
+    | 'The Beatles' | 'Here Comes the Sun'        |
+    When I search for text {'door'}
+    Then I see songs
+    | 'artist'      | 'name'                      |
+    | 'The doors'   | 'Riders on the storm'       |
+    | 'Bob dylan'   | "Knockin' On Heaven's Door" |
```
Comment (olexale) [inline README.md:143]:
```suggestion
    | 'The Doors'   | 'Riders on the storm'       |
    | 'Bob Dylan'   | "Knockin' On Heaven's Door" |
```

File: README.md [hunk 3]
```diff
@@ -125,6 +125,57 @@ Feature: Sample
...
+```
+For each of the above step lines that are followed by a table, in the related generate step file, the created function will have an object parameter of type `DataTable`:
+```dart
+import 'package:bdd_widget_test/data_table.dart' as bdd;
+import 'package:flutter_test/flutter_test.dart';
+
+/// Usage: the following songs
```
Comment (olexale) [inline README.md:150]:
```suggestion
/// Usage: Given available songs
```

File: README.md [hunk 4]
```diff
@@ -125,6 +125,57 @@ Feature: Sample
...
+We can then iterate that list and extract data from each map using the "column" header as key.
+For the above example, the asMaps function returns:
+```dart
+[
+    {'artist': 'The doors', 'name': 'Riders on the storm'},
+    {'artist': 'Bob dylan', 'name': "Knockin' On Heaven's Door"},
+    {'artist': 'The Beatles', 'name': 'Here Comes the Sun'},
+]
+```
+**In this case, we must provide a heading for our table.**
```
Comment (olexale) [inline README.md:177]: I'd suggest keeping the documentation brief as that would increase the chances that someone will read it. Do you think the following description might be sufficient?
```suggestion
Use the `DataTable` parameter to get access to the data:
```dart
final dataAsList = dataTable.asLists(); // [['artist', 'name'], ['The Doors', 'Riders on the storm'], ...]
final dataAsMaps = dataTable.asMaps(); // [{'artist: 'The Doors', 'name: 'Riders on the storm'}, ...]
```
```

Comment (ron-brosh) [reply README.md:177]: haha I might got too excited with the documentation 😄 
Sure, less is more in this context!

**Comment by olexale** [comment]
Thanks, Ron! I'm publishing a new version now 🙂

**Pull request merged**